File: py/Euler86.py
def is_square(n):
    return n**0.5 == int(n**0.5)

def count(a, b, c):
    s = a**2 + (b+c)**2
    if is_square(s): 
        return 1
    return 0

m = 0
cnt = 0

while cnt < 1000000:
    m += 1
    a = m
    for b in range(1, m+1):
        for c in range(b, m+1):
            cnt += count(a, b, c)
    print(cnt)

print(m, cnt)

File: c/Euler86.c
#include <stdio.h>
#include <math.h>

int is_squre(long n) {
    return sqrt(n) == (int)(sqrt(n));
}

int count(long a, long b, long c) {
    long s = pow(a, 2) + pow(b+c, 2);
    if (is_squre(s))
        return 1;
    return 0;
}

int main() {
    int m = 0;
    int cnt = 0;

    while (cnt < 1000000) {
        m++;
        int a = m;
        for (int b = 1; b < m+1; b++) {
            for (int c = b; c < m+1; c++) {
                cnt += count(a, b, c);
            }
        }
        printf("%d\n", cnt);
    }
    printf("%d %d\n", m, cnt);
}